scripts/BUILD.bazel

```diff
@@ -42,3 +43,19 @@ kt_jvm_binary(
         "//scripts/src/java/org/oppia/android/scripts/ci:compute_affected_tests_lib",
     ],
 )
+
+REGEX_PATTERN_CHECK_ASSETS = retrieve_regex_pattern_check_assets("regex_pattern_assets")
+
+kt_jvm_library(
+    name = "regex_check_assets",
```

Comment:
Could these be defined under scripts/.../regex? That seems like a better place since it's another library to include in both tests & the binary.

Reply:
Ah! actually, this was the main issue. The pipeline will not work here because it looks for the assets at the location where it is called from the BUILD.bazel file, and since the assets are present at the root level of scripts and if we shift this to scripts/.../regex, then it actually fails to find those, as the assets are present some levels up. Therefore, it looks like we can't define it at scripts/.../regex

Reply:
Why can't we make the macros use a relative path from the root of the Bazel repository to the locations of the textproto files? I don't understand why any of the current code would require things to be in the same directory.

Reply:
Actually, I tried shiftng everything under scripts/.../regex. Doing this results in this error:

`GenerateBinaryProtoFromText scripts/src/java/org/oppia/android/scripts/regex/assets/filename_pattern_validation_checks.pb failed: missing input file '//scripts/src/java/org/oppia/android/scripts/regex:assets/filename_pattern_validation_checks.textproto'`

I have pushed all the new changes temporarily PTAL.

We could rectify this by removing the `regex_assets.bzl` file and keep everything in the `script_assets.bzl`. Just like it is done currently in the `domain` layer.
This will maintain uniformity for all the coming scripts.

Reply:
Also, I tried giving a relative path: `//scripts/assets`.
This gives the following error:
`ERROR: /home/dell/Desktop/staticanalysis/oppia-android/scripts/BUILD.bazel:47:65: //scripts:generate_binary_proto_for_text_proto_regex_pattern_assets_file_content_validation_checks: label '//scripts/assets/file_content_validation_checks.pb' is not in the current package`